common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java

```diff
@@ -128,23 +129,13 @@ public List<ShuffleDataSegment> split(ShuffleIndexResult shuffleIndexResult) {
         }
 
         if (expectTaskIds.contains(taskAttemptId)) {
-          if (bufferOffset != 0 && index - lastExpectedBlockIndex > 1) {
```

Comment:
Why do we remove this?

Reply:
This checking logic is to check the discontinuous blocks in a ordered sequence. So this will throw exception when encountering the discontinuous task map ids

Reply:
I got it.